content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -238,55 +219,44 @@ Now you see, we’ve created a new dog with the name Bingo. You can go on to cre
+  Dog barking...
  
-Here's another example to make things clearer: imagine a car driver. He is only interested in starting, moving, and stopping the car, speed, or change gear and brake, but he is not interested in the mechanism that allows him to do this. In abstraction, we hide the implementation details from the user and expose the functionality.
+  Is the dog barking:? true
  
-This way, the user will only know what a function(method in java) does but not how it does it.
+  Eating
+ 
+  Is the dog eating:? true
  
-### How to implement Abstraction in Java:
+  The dog name is: Bingo
+ 
+Now that we’ve got this out of the way let's talk about abstraction and encapsulation.
+ 
+### Abstraction 
+Who could have imagined that abstraction will be a thing in programming? Who? But right now, you can’t talk about OOP without mentioning abstraction. It is a significant building block of OOP. So then, what is abstraction? This section will explain what abstraction is and how to use it as a Java developer.
+
+In abstraction, only the useful attributes of an object are on display or accessible. The low-level details are “hidden”. For example, while you’re typing on your computer. You’re interested in punching the keys believing that it produces results on your screen. You know nothing about the mechanism that produces the result, and I’m sure you haven’t even thought about it yet. Have you?
  
-    In Java, abstraction is implemented in either of these two ways:
-    
-    - By using the non-access modifier ‘abstract’ keyword or
-    
-    - Interface
-    
-    - Non-access modifier keyword “Abstract” implementation
+Here's another example to make things clearer: imagine car drivers. They are only interested in starting, moving, and stopping the car, speed, or changing gear and brake, but they are not interested in the mechanism that allows them to do this. In abstraction, we hide the implementation details from the user while exposing the functionality.
  
-You can only use the abstract access modifier keyword with classes and methods but not with variables. Methods with the keyword abstract are methods without implementation.
+This way, the user will only know what a function(method in java) does but not how it does it.
  
-To implement an abstract class in Java, use the abstract keyword, e.g., `public abstract class Circle`, where Circle is the class name.
+### How to implement Abstraction in Java
+In Java, abstraction is implemented using either of these two ways:
+- By using the non-access modifier ‘abstract’ keyword or
+- Interface
  
-Use this pattern for abstract method implementation, `public abstract returnType methodName()`.
+- Non-Access modifier keyword “Abstract” implementation
```

Comment:
> - Non-Access modifier keyword “Abstract” implementation

I'm confused about why you made this a list item. It seems to me your intention was to make this a subheader. I would do that using four \#s like so:

> \#\#\#\# Non-Access modifier keyword “Abstract” implementation

Also, I don't know what you are trying to say with this subheader, it's a little confusing. I think maybe a more appropriate subheader would be:

> \#\#\#\# Working with the Abstract Keyword